platform/openide.util.ui/src/org/openide/util/ImageUtilities.java

```diff
@@ -1127,23 +1068,7 @@ public String toString() {
     /**
      * Key used for ToolTippedImage
      */
-    private static class ToolTipImageKey {
-        Image image;
-        String str;
-
-        ToolTipImageKey(Image image, String str) {
-            this.image = image;
-            this.str = str;
-        }
-
-        @Override
-        public boolean equals(Object other) {
-            if (!(other instanceof ToolTipImageKey)) {
-                return false;
-            }
-            ToolTipImageKey k = (ToolTipImageKey) other;
-            return (str.equals(k.str)) && (image == k.image);
-        }
+    private record ToolTipImageKey(Image image, String str) {
```

Comment:
Same thing here; we should not define a custom hashCode when we are using record, no?

Reply:
reverted the removed equals - it should make sense again.

Records are just classes with exclusively final fields which are all mapped to constructor parameters. Its fine to override equals/hashcode/tostring. Sorry for adding confusion by removing the equals.